frontend/app/src/jarvis/main/body.tsx

```diff
@@ -144,9 +144,18 @@ export default function BodyComponent() {
   }, DEBOUNCE_LAG), [])
 
   const onChange = (event: any) => {
-    setTextBoxText(event.target.value)
+    setTopic(event.target.value)
     dispatchNetwork(event.target.value)
   }
+
+  const onNodeClicked = (event: PointerEvent, data: any) => {
+    if (data.type === 'topic') {
+      if (!isLoading) {
+        setTopic(data.name)
+        callApi(data.name)
```

Comment:
I think we can change this to just `onChange(data.name)` since I think it does both `setTopic` and `callApi` aswell

also on replit I'm noticing that `isLoading` is always true for some reason even though I see `Running finally block` in the console output

Reply:
Yes, this is the issue. I was running a mockup project with react 18 and some other dependencies changed. Maybe that's the reason why it doesn't work here. I'll try a workaround.

Reply:
I made some changes in the replit but not sure if that's how you want to do it lemme know what you think

do you know what dependency change was causing the issue? Are we just not able to update state in the finally block?

Reply:
And regarding the `onChange` use, currently `onChange` & `onNodeClicked` receive different types of events. But sure, we could extract the topic before and pass it cleanly.

Reply:
yup exactly what I was thinking 🚀